src/System.Net.Security/src/System/Net/ContextFlagsAdapterPal.Unix.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace System.Net
+{
+    internal static class ContextFlagsAdapterPal
+    {
+        private struct ContextFlagMapping
+        {
+            public readonly Interop.NetSecurityNative.GssFlags GssFlags;
+            public readonly ContextFlagsPal ContextFlag;
+
+            public ContextFlagMapping(Interop.NetSecurityNative.GssFlags gssFlag, ContextFlagsPal contextFlag)
+            {
+                GssFlags = gssFlag;
+                ContextFlag = contextFlag;
+            }
+        }
+
+        private static readonly ContextFlagMapping[] s_contextFlagMapping = new[]
+        {
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_INTEG_FLAG, ContextFlagsPal.AcceptIntegrity | ContextFlagsPal.InitIntegrity),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_CONF_FLAG, ContextFlagsPal.Confidentiality),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_IDENTIFY_FLAG, ContextFlagsPal.InitIdentify),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_MUTUAL_FLAG, ContextFlagsPal.MutualAuth),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_REPLAY_FLAG, ContextFlagsPal.ReplayDetect),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_SEQUENCE_FLAG, ContextFlagsPal.SequenceDetect)
+        };
+
+
+        internal static ContextFlagsPal GetContextFlagsPalFromInterop(Interop.NetSecurityNative.GssFlags gssFlags)
+        {
+            ContextFlagsPal flags = ContextFlagsPal.Zero;
+            foreach (ContextFlagMapping mapping in s_contextFlagMapping)
+            {
+                if ((gssFlags & mapping.GssFlags) != 0)
+                {
+                    flags |= mapping.ContextFlag;
+                }
+            }
+
+            return flags;
+        }
+
+        internal static Interop.NetSecurityNative.GssFlags GetInteropFromContextFlagsPal(ContextFlagsPal flags)
+        {
+            Interop.NetSecurityNative.GssFlags gssFlags = (Interop.NetSecurityNative.GssFlags)0;
+            foreach (ContextFlagMapping mapping in s_contextFlagMapping)
+            {
+                if ((flags & mapping.ContextFlag) != 0)
```

Comment:
This is going to map in GSS_C_INTEG_FLAG for either AcceptIntegrity or InitIntegrity (not requiring them both to be set).  If you need them both to be set, this should be == mapping.ContextFlag instead of != 0.  If "either" is what you want, it might be worthy of a comment.  (And if "both" is what you want, that might also warrant a comment :))